scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1320,6 +1401,9 @@ def __init__(
         self._pod_processor = PodProcessor(self._controllers)
         self._pods_cache = _K8sCache(self._pod_processor, "Pod")
 
+        # create the namespace cache
+        self._namespace_cache = _K8sCache(NamespaceProcessor(self._controllers), "Namespace")
```

Comment:
Any idea how also caching the namespace information affects the agent memory consumption on larger clusters?

I imagine that overhead is not so big since there will usually be much less namespaces than Pods (which we already cache).

Reply:
That was my thinking. The NamespaceInfo object holds only necessary parts of metadata about the object and gets purged when expired.